### ISSUE TITLE
`custom-error-definition`: rule silently skips constructors whose first parameter is a TypeScript parameter property

### ISSUE THREAD

**Issue by lexa-uw** [sindresorhus/eslint-plugin-unicorn#3615]
When the first constructor parameter is declared as a TypeScript parameter property `(public readonly x: string / private readonly x: string)`, the rule returns before running any of its checks. Such classes are never reported — not for a missing `options` parameter, not for a missing `this.name`, not for anything else. They look compliant in a green lint run, but they simply were not examined.

```typescript
// Reported as expected — plain identifier parameter
export class PlainError extends Error {
  public constructor(requestId: string) {
    super(`Result for id "${requestId}" does not exist`)
    this.name = 'PlainError'
  }
}
// error: Error constructors should accept `options` as the second parameter.

// NOT reported — identical semantics, only the parameter form differs
export class ParameterPropertyError extends Error {
  public constructor(public readonly requestId: string) {
    super(`Result for id "${requestId}" does not exist`)
    this.name = 'ParameterPropertyError'
  }
}
// no error — the native `cause` chain is just as unavailable here
```

Version: eslint-plugin-unicorn 73.0.0
